ptf/tests/config/basic_config.py

```diff
@@ -47,11 +112,15 @@ def get_port_list(self, test):
         Output variable:
             test.port_list
         """
-        port_list = sai_thrift_object_list_t(count=100)
+        port_list = sai_thrift_object_list_t(idlist=[], count=100)
         p_list = sai_thrift_get_switch_attribute(
             test.client, port_list=port_list)
         test.port_list = p_list['port_list'].idlist
 
+        #Gets self.portX objects for all active ports
+        for i, _ in enumerate(test.port_list):
+            setattr(test, 'port%s' % i, test.port_list[i])
```

Comment:
there should be only one kind of variable stand for the collection data, like port, hostif. 
Please choose naming it as port_NUM, or post_list, then we can follow yours.